packages/nimble-components/src/table/index.ts

```diff
@@ -1081,13 +1081,10 @@ export class Table<
         const slotsByRecordId = this.getRequestedSlotsByRecordId();
         this.tableData = rows.map(row => {
             const isGroupRow = row.getIsGrouped();
-            const hasParentRow = isGroupRow ? false : row.getParentRow();
             const isParent = !isGroupRow && this.getRowCanExpand(row);
-            const isChildOfGroupRowWithNoHierarchy = !isGroupRow
-                && !isParent
-                && !hasParentRow
-                && row.depth > 0
-                && !this.parentIdFieldName;
+            const isDataChildOfGroupRowWithNoHierarchy = !isGroupRow // is a data row (not a group row)
+                && !this.parentIdFieldName // table does not have hierarchy enabled
```

Comment:
We have `isHierarchyEnabled` for this check (seems like it's used inconsistently, worth a consistency scan?)